storage/innobase/handler/ha_innodb.cc

```diff
@@ -15950,42 +16016,67 @@ static int innobase_alter_encrypt_tablespace(handlerton *hton, THD *thd,
       !(oldenc.empty() || Encryption::is_none(oldenc.data()));
   /* If new tablespace definition says it's encrypted */
   bool is_new_encrypted = !Encryption::is_none(newenc.data());
+  fil_space_crypt_t *crypt_data = space->crypt_data;
 
   // It is only possible to mark tablespace to be skipped by encryption threads
   // if it is not already encrypted.
   if (!is_old_encrypted && !is_new_encrypted &&
       alter_info->explicit_encryption) {
```

Comment:
old is not encrypted and new is not encrypted but expicit_encryption is set? Can you tell me how? ALTER TABLESPACE ENGINE=INNODB?

Reply:
For instance tablespace was created without any ENCRYPTION clause - CREATE TABLESPACE ... - then the user decided to explicitly remove this tablespace from encryption threads by specifying ALTER TABLESPACE ENCRYPTION='N'. So before and after is not encrypted, but explicit_encryption is set. Probably the name is misleading explicit_encryption - as it was explicitly encrypted, whilst it means that ENCRYPTION clause was explicitly specified. Do you think we should change the name ? Although I do not know if this would not break the compatibility and we would need to rename it at the upgrade, bit painful ...

Reply:
Ok, lets keep the name. Is there test that covers this change? Also will DD attribute 'explicit_encrytion' be present ?

Reply:
yes, explicit_encryption in DD should be set when you use explicit clause.